### PR TITLE
boards: Rename pca20035 to Thingy91

### PR DESCRIPTION
**Pull request by jhn-nordic**
Updated files to use new board name format

Signed-off-by: Jon Helge Nistad <jon.helge.nistad@nordicsemi.no>

### CODE REVIEW

File: doc/release-notes-1.1.0.rst [hunk 1]
```diff
@@ -230,9 +230,9 @@ BSD library
 Board support
 -------------
 
-* Thingy:91 (``nrf9160_pca20035``):
+* Thingy:91 (``thingy91_nrf9160``):
 
-  *  Removed support for earlier hardware versions of ``nrf52840_pca20035`` and ``nrf9160_pca20035``.
+  *  Removed support for earlier hardware versions of ``thingy91_nrf52840`` and ``thingy91_nrf9160``.
```
Comment (ioannisg) [inline doc/release-notes-1.1.0.rst:235]: Do we need to "retroactively" change release notes?

Comment (jhn-nordic) [reply doc/release-notes-1.1.0.rst:235]: Agree. Victim of search and replace.

File: doc/nrf/shortcuts.txt [hunk 2]
```diff
@@ -23,7 +23,7 @@
 .. |nRF51DK| replace:: nRF51 DK board (PCA10028)
 .. |nRF52DK| replace:: nRF52 DK board (PCA10040)
 .. |nRF52840Dongle| replace:: nRF52840 Dongle (PCA10059)
-.. |Thingy91| replace:: Thingy:91 (PCA20035) - see :ref:`ug_thingy91`
+.. |Thingy91| replace:: Thingy:91 (THINGY91) - see :ref:`ug_thingy91`
```
Comment (ru-fu) [inline doc/nrf/shortcuts.txt:26]: This should stay as is - the PCA number is still correct, and here we are not referring to the build target.

File: doc/nrf/ug_thingy91.rst [hunk 3]
```diff
@@ -197,7 +197,7 @@ Building and programming using SEGGER Embedded Studio
 .. figure:: images/ses_thingy_configuration.png
```
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst:197]: This screenshot must be updated.

File: doc/release-notes-1.1.0.rst [hunk 4]
```diff
@@ -46,7 +46,7 @@ Supported boards
 ****************
 
 * PCA10090 (nRF9160 DK)
-* PCA20035 (Thingy:91)
+* THINGY91 (Thingy:91)
```
Comment (ru-fu) [inline doc/release-notes-1.1.0.rst:49]: This should stay as is.

File: doc/release-notes-1.1.0.rst (hunk 1 above)
Comment (ru-fu) [reply doc/release-notes-1.1.0.rst:235]: Correct, this shouldn't be updated (or only if there is a reference that breaks, but I don't think that will be the case for the Thingy).

File: doc/release-notes-1.2.0.rst [hunk 5]
```diff
@@ -47,7 +47,7 @@ Tested boards
 The following boards have been used during testing of this release:
 
 * PCA10090 (nRF9160 DK)
-* PCA20035 (Thingy:91)
+* THINGY91 (Thingy:91)
```
Comment (ru-fu) [inline doc/release-notes-1.2.0.rst:50]: Keep as is.

File: samples/sensor/bh1749/README.rst [hunk 6]
```diff
@@ -30,7 +30,7 @@ Building and Running
 This project outputs sensor data to the console. It requires a BH1749
 sensor. It should work with any platform featuring a I2C peripheral interface.
 It does not work on QEMU.
-In this example below the PCA20035 board is used.
+In this example below the THINGY91 board is used.
```
Comment (ru-fu) [inline samples/sensor/bh1749/README.rst:33]:
```suggestion
The example below uses the Thingy:91.
```

**Comment by thst-nordic** [comment]
TEST_IOT_THINGY91 system test repo needs a search-and-replace also.

File: boards/arm/thingy91_nrf52840/thingy91_nrf52840.dts [hunk 7]
```diff
@@ -8,8 +8,8 @@
 #include <nordic/nrf52840_qiaa.dtsi>
 
 / {
-	model = "Nordic PCA20035";
-	compatible = "nordic,pca20035", "nordic,nrf52840-qiaa",
+	model = "Nordic THINGY91";
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf52840/thingy91_nrf52840.dts:11]: This is the format used for Thingy:52 upstream now: `model = "Nordic Thingy52 NRF52832";`

File: applications/connectivity_bridge/src/main.c [hunk 8]
```diff
@@ -14,9 +14,9 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(MODULE);
 
-#define USB_SERIALNUMBER_TEMPLATE "PCA20035_%04X%08X"
+#define USB_SERIALNUMBER_TEMPLATE "THINGY91_%04X%08X"
 
-static u8_t usb_serial_str[] = "PCA20035_12PLACEHLDRS";
+static u8_t usb_serial_str[] = "THINGY91_12PLACEHLDRS";
```
Comment (jtguggedal) [inline applications/connectivity_bridge/src/main.c:19]: @wlgrd Will this renaming cause issues for productiont test?

File: boards/arm/thingy91_nrf52840/Kconfig [hunk 9]
```diff
@@ -1,14 +1,14 @@
-# Kconfig - nRF52840 PCA20035 board configuration
+# nRF52840 THINGY91 board configuration
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf52840/Kconfig:1]: Perhaps `Thingy:91 nRF52840` instead. Same also other places and for the onboard nRF9160

File: boards/arm/thingy91_nrf52840/thingy91_nrf52840.yaml [hunk 10]
```diff
@@ -1,5 +1,5 @@
-identifier: nrf52840_pca20035
-name: nRF52840-PCA20035
+identifier: thingy91_nrf52840
+name: THINGY91-nRF52840
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf52840/thingy91_nrf52840.yaml:2]: Format for Thingy:52: `name: Thingy52-NRF52832`

File: boards/arm/thingy91_nrf9160/CMakeLists.txt [hunk 11]
```diff
@@ -1,15 +1,15 @@
-# Kconfig - nRF91 PCA20035 board configuration
+# Kconfig - nRF91 THINGY91 board configuration
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf9160/CMakeLists.txt:1]: Same as above, perhaps `Thingy:91 nRF9160` instead

File: boards/arm/thingy91_nrf9160/thingy91_nrf9160.yaml [hunk 12]
```diff
@@ -1,5 +1,5 @@
-identifier: nrf9160_pca20035
-name: nRF9160-PCA20035
+identifier: thingy91_nrf9160
+name: THINGY91-nRF9160
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf9160/thingy91_nrf9160.yaml:2]: Suggest `name: Thingy91-NRF9160` To align with Thingy:52. Or actually, `name: Thingy91-nRF9160` looks slightly better..

File: lib/lte_link_control/lte_lc.c [hunk 13]
```diff
@@ -504,7 +504,7 @@ static int w_lte_lc_init(void)
 	}
 
 #if defined(CONFIG_LWM2M_CARRIER) && !defined(CONFIG_GPS_USE_SIM) && \
-    (defined(CONFIG_BOARD_NRF9160_PCA20035NS) || \
+    (defined(CONFIG_BOARD_THINGY91_NRF9160NS) || \
      defined(CONFIG_BOARD_NRF9160_PCA10090NS))
```
Comment (jtguggedal) [inline lib/lte_link_control/lte_lc.c:508]: The below line (and similar elsewhere) should also be changed at some point. Probably not in this PR though

**Comment by jhn-nordic** [comment]
@ioannisg @hakonfam @jtguggedal  I believe I have addressed your comments now.

File: boards/arm/thingy91_nrf9160/Kconfig.board [hunk 14]
```diff
@@ -1,17 +1,17 @@
-# Kconfig - nRF91 PCA20035 board configuration
+# nRF91 THINGY91 board configuration
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf9160/Kconfig.board:1]: Should this line be changed as well to match the `Thingy:91 nRF9160 board configuration` above?

File: boards/arm/thingy91_nrf9160/Kconfig.defconfig [hunk 15]
```diff
@@ -1,13 +1,13 @@
-# Kconfig - nRF91 PCA20035 board configuration
+# nRF91 THINGY91 board configuration
```
Comment (jtguggedal) [inline boards/arm/thingy91_nrf9160/Kconfig.defconfig:1]: Same as above

File: doc/nrf/ug_thingy91.rst (hunk 3 above)
Comment (jhn-nordic) [reply doc/nrf/ug_thingy91.rst:205]: can we just "photoshop" it?

Comment (jhn-nordic) [reply doc/nrf/ug_thingy91.rst:205]: added a new screenshot

Comment (ru-fu) [reply doc/nrf/ug_thingy91.rst:205]: The new one has thingy91_nrf9160 as target, not thingy91_nrf9160ns - is that correct?

Comment (jonhelge) [reply doc/nrf/ug_thingy91.rst:205]: No. I will fix

Comment (jhn-nordic) [reply doc/nrf/ug_thingy91.rst:205]: new image with *ns is pushed now

**Comment by jtguggedal** [comment]
Looks like it might need a rebase to pass CI

**Comment by jhn-nordic** [comment]
Updated west.yml to point at https://github.com/NordicPlayground/fw-nrfconnect-mcuboot/commit/19dce6e3034ac546dba03d523cbd0f7784f5e0e2

**Comment by jhn-nordic** [comment]
> Some additional feedback, see comments.
> Also I've not found a doc sentence that stresses the board renames. See how it is done in the upstream project (example from nRF9160 DK):
> 
> ```
> .. note::
> 
>    In previous Zephyr releases this board was named *nrf9160_pca10090*.
> ```

It is a part of the changelog so I expect it will be written in the release notes for 1.3.0

File: cmake/boilerplate.cmake [hunk 16]
```diff
@@ -7,6 +7,8 @@
 # Point to NCS root directory.
 set(NRF_DIR ${CMAKE_CURRENT_LIST_DIR}/..)
 
+include(${NRF_DIR}/boards/deprecated.cmake)
```
[the remark this answers is not in the document]
Comment (tejlmand) [reply cmake/boilerplate.cmake:10]: This looks fine.
It follows the same principle as here: https://github.com/NordicPlayground/fw-nrfconnect-zephyr/blob/master/boards/deprecated.cmake
and as this list contains only downstream board, everything looks good.

**Comment by ioannisg** [comment]
> > Some additional feedback, see comments.
> > Also I've not found a doc sentence that stresses the board renames. See how it is done in the upstream project (example from nRF9160 DK):
> > ```
> > .. note::
> > 
> >    In previous Zephyr releases this board was named *nrf9160_pca10090*.
> > ```
> 
> It is a part of the changelog so I expect it will be written in the release notes for 1.3.0

Well this is not enough if we would like to be consistent with the upstream board renaming policy. In Zephyr we have done all three things:
- adding the board renaming in the release notes for next Zephyr release
- adding CMake entries for backwards compatibility
- adding these minimal notes in the .rst docs of the boards (what is still missing here)

Any reason we don't want to fully follow the upstream solution?

File: doc/nrf/ug_thingy91.rst [hunk 17]
```diff
@@ -151,14 +151,18 @@ The build targets of interest for Thingy:91 in |NCS| are as follows:
...
-|nRF52840 SoC   |``nrf52840_pca20035``                             |
+|nRF52840 SoC   |``thingy91_nrf52840``                             |
 +---------------+--------------------------------------------------+
 
-You should use the build target ``nrf9160_pca20035`` or ``nrf9160_pca20035ns`` when building the application code for the nRF9160 SiP and the build target ``nrf52840_pca20035`` when building the application code for the onboard nRF52840 SoC.
+You should use the build target ``thingy91_nrf9160`` or ``thingy91_nrf9160ns`` when building the application code for the nRF9160 SiP and the build target ``thingy91_nrf52840`` when building the application code for the onboard nRF52840 SoC.
+
+.. note::
+
+   In previous NCS releases these build targets were named *nrf9160_pca20035* and *nrf52840_pca20035*.
```
Comment (jhn-nordic) [inline doc/nrf/ug_thingy91.rst:165]: @ioannisg

**Comment by jhn-nordic** [comment]
> > > Some additional feedback, see comments.
> > > Also I've not found a doc sentence that stresses the board renames. See how it is done in the upstream project (example from nRF9160 DK):
> > > ```
> > > .. note::
> > > 
> > >    In previous Zephyr releases this board was named *nrf9160_pca10090*.
> > > ```
> > 
> > 
> > It is a part of the changelog so I expect it will be written in the release notes for 1.3.0
> 
> Well this is not enough if we would like to be consistent with the upstream board renaming policy. In Zephyr we have done all three things:
> 
> * adding the board renaming in the release notes for next Zephyr release
> * adding CMake entries for backwards compatibility
> * adding these minimal notes in the .rst docs of the boards (what is still missing here)
> 
> Any reason we don't want to fully follow the upstream solution?

I did not understand your point until now. Should be fixed now

File: doc/nrf/ug_thingy91.rst (hunk 17 above)
Comment (ioannisg) [reply doc/nrf/ug_thingy91.rst:165]: Fine with me.

File: applications/connectivity_bridge/src/main.c (hunk 8 above)
Comment (wlgrd) [reply applications/connectivity_bridge/src/main.c:19]: We'll manage :)

File: doc/nrf/ug_thingy91.rst (hunk 17 above)
Comment (ru-fu) [inline doc/nrf/ug_thingy91.rst:165]:
```suggestion
   In |NCS| releases before v1.3.0, these build targets were named ``nrf9160_pca20035``, ``nrf9160_pca20035ns``, and ``nrf52840_pca20035``.
```

**Pull request merged**